brat/Brat/Checker/Monad.hs

```diff
@@ -60,8 +60,8 @@ data Context = Ctx { globalVEnv :: VEnv
 
 data CheckingSig ty where
   Fresh   :: String -> CheckingSig Name
-  -- Run a sub-process on a new namespace-level
-  InLvl   :: String -> Checking a -> CheckingSig a
+  GetNS   :: CheckingSig Namespace
```

Comment:
How about a single `SplitNS :: String -> CheckingSig Namespace` which both updates the namespace inside `handler` (to the second half of the split) and returns the first half of the split?

Reply:
This makes things much nicer! 😄